app/javascript/tubes-to-rack/components/TubesToRack.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+  <lb-page>
+    <lb-loading-modal
+      v-if="loading"
+      :message="progressMessage"
+    />
+    <lb-main-content>
+      <b-card
+        bg-variant="dark"
+        text-variant="white"
+      >
+        <h3>Tube rack UI goes here</h3>
+        <p>
+          Scanned tubes will appear in a UI representation of the tube rack.
+          This will be similar in function to the plate UI but will be capable
+          of opening the tube information page when viewed after creation.
+        </p>
+      </b-card>
+    </lb-main-content>
+    <lb-sidebar>
+      <b-card
+        header="Scan tubes"
+        header-tag="h3"
+      >
+        <b-form-group
+          label="Scan the tube barcodes into the relevant rack coordinates:"
+          style="position:relative; height:460px; overflow-y:scroll;"
+        >
+          <lb-labware-scan
+            v-for="i in tubeCount"
+            :key="i"
+            :api="devourApi"
+            :label="indexToName(i - 1)"
+            :fields="tubeFields"
+            :includes="tubeIncludes"
+            :validators="scanValidators"
+            :colour-index="i"
+            :labware-type="'tube'"
+            :valid-message="''"
+            @change="updateTube(i, $event)"
+          />
+        </b-form-group>
+        <hr>
+        <b-button
+          :disabled="!valid"
+          variant="success"
+          @click="createRack()"
+        >
+          Create
+        </b-button>
+      </b-card>
+    </lb-sidebar>
+  </lb-page>
+</template>
+
+<script>
+import devourApi from 'shared/devourApi'
+import filterProps from './filterProps'
+import LabwareScan from 'shared/components/LabwareScan'
+import LoadingModal from 'shared/components/LoadingModal'
+import resources from 'shared/resources'
+import { buildTubeObjs } from 'shared/tubeHelpers'
+import { checkDuplicates, checkMatchingPurposes } from 'shared/components/tubeScanValidators'
+
+export default {
+  name: 'TubesToRack',
+  components: {
+    'lb-labware-scan': LabwareScan,
+    'lb-loading-modal': LoadingModal
+  },
+  props: {
+    // Sequencescape API V2 URL
+    sequencescapeApi: { type: String, default: 'http://localhost:3000/api/v2' },
+
+    // Number of source tubes
+    tubeCount: { type: Number, default: 16 }
```

Comment:
Combined with some of the other thoughts, it may be better to pass in the rack dimensions, and calculate the size. (I've been going with numberOfRows and numberOfColumns where I can, as it feels less ambiguous than height and width)

I'll also briefly ask how you're planning to deal with the tube-rack purpose? There are currently 3 registered in SS, two 96 size (or which we probably only want to care about TR96) and a 48 well size (TR48). Typically the purpose is passed to the controller as a parameter, which means the page itself will already know what it's creating. Happy to come back to this one though, as there's still a little glue needed to make the buttons also appear on the tube pages.